slasher/detection/attestations/spanner.go

```diff
@@ -197,12 +213,9 @@ func (s *SpanDetector) updateMaxSpan(ctx context.Context, source uint64, target
 				SigBytes:    span.SigBytes,
 				HasAttested: span.HasAttested,
 			}
-			err := s.slasherDB.SaveValidatorEpochSpans(ctx, valIdx, epoch, span)
-			if err != nil {
+			if err := s.slasherDB.SaveValidatorEpochSpans(ctx, valIdx, epoch, span); err != nil {
 				return err
 			}
-		} else {
-			break
```

Comment:
I made this change because I thought it could cause an issue and not properly update spans, wdyt?